ast/src/analyzed/mod.rs

```diff
@@ -841,6 +842,13 @@ impl From<&Symbol> for PolyID {
     }
 }
 
+impl Hash for PolyID {
+    fn hash<H: Hasher>(&self, state: &mut H) {
+        // Only use `id` for hashing
```

Comment:
If we do this, the hash gets incompatible with Eq, and we should take extra care never to put any polynomials of different kinds in the same hash set/map. Can we guarantee that?

Reply:
Oh right! Can you add a quick comment about that?

Reply:
ok, so actually did it a bit differently to keep the old behavior and save the extra hash call